charts/gateway/templates/tests/test-connection.yaml

```diff
@@ -38,13 +38,26 @@ spec:
           mountPath: /etc/persistence
   containers:
     - name: wget
-      image: busybox
+      image: alpine
+      envFrom:
+        - secretRef:
+            name: node-ip
       # Get the list of backends, which should return an empty list, "[]". For this test to pass
       # the gateway must successfully connect to an initialized backend database
       command:
         - "sh"
         - "-c"
-        - '[ "$(wget {{ include "trino-gateway.fullname" . }}:{{ .Values.service.port }}/entity/GATEWAY_BACKEND -O -)" = "[]" ]'
+        - |
+          [ 1 = 1 ]
+          {{- if eq .Values.service.type "NodePort" -}}
+             && [ "$(wget --no-check-certificate https://${NODE_IP}:30443/entity/GATEWAY_BACKEND -O -)" = "[]" ] && [ "$(wget http://${NODE_IP}:30080/entity/GATEWAY_BACKEND -O -)" = "[]" ]
+          {{- end }}
+          {{- if index .Values "config" "serverConfig" "http-server.https.enabled" -}}
+             && [ "$(wget --no-check-certificate https://trino-gateway:8443/entity/GATEWAY_BACKEND -O -)" = "[]" ]
+          {{- end }}
+          {{- if index .Values "config" "serverConfig" "http-server.http.enabled" -}}
```

Comment:
Does this check if the value is true, or is the property set?

Reply:
it checks the truth value of the property. I double checked by adding 
```
    http-server.https.enabled: false
```
to one of the non-https tests, and it passes.